staging/src/k8s.io/client-go/dynamic/golden_test.go

```diff
@@ -246,3 +249,157 @@ func TestGoldenRequest(t *testing.T) {
 		})
 	}
 }
+
+type RoundTripperFunc func(*http.Request) (*http.Response, error)
+
+func (f RoundTripperFunc) RoundTrip(r *http.Request) (*http.Response, error) {
+	return f(r)
+}
+
+// TestGoldenResponse tests that the objects returned from dynamic client methods, given a fixed
+// HTTP response, are not changed unintentionally by changes to the client.
+func TestGoldenResponse(t *testing.T) {
```

Comment:
reminder to myself that we already did this on the request side, so now we're checking to be sure that client is properly deserializing and decoding the response.